src/main/scala/com/typesafe/sbt/packager/archetypes/jlink/JlinkPlugin.scala

```diff
@@ -176,12 +176,26 @@ object JlinkPlugin extends AutoPlugin {
   private lazy val defaultJavaHome: File =
     file(sys.props.getOrElse("java.home", sys.error("no java.home")))
 
-  private def runJavaTool(jvm: File, log: Logger)(exeName: String, args: Seq[String]): ProcessBuilder = {
-    val exe = (jvm / "bin" / exeName).getAbsolutePath
+  private def runJavaTool(jvm: File, log: Logger)(toolName: String, args: Seq[String]): String = {
+    log.info("Running: " + (toolName +: args).mkString(" "))
 
-    log.info("Running: " + (exe +: args).mkString(" "))
+    val toolLauncherClass = classOf[ru.eldis.toollauncher.ToolLauncher]
+    val toolLauncherJar = file(
+      // This assumes that the code source is a file or a directory (as opposed
+      // to a remote URL) - but that should hold.
+      toolLauncherClass.getProtectionDomain.getCodeSource.getLocation.getPath
+    ).getAbsolutePath
```

Comment:
Just if I got this right:
We summon the `ToolLauncher` class to get the path to the `jar` file, which we require to run it with `Process`

Reply:
Yes, that is correct. We could resolve the JAR separately (via `libraryDependencies`, or directly via `DependencyResolution`), but it's easier to just bring it as a plugin dependency. It has no transitive dependencies, so it shouldn't break anything here.